config/samples/backends/pure/nvme-roce/backend.yaml

```diff
@@ -0,0 +1,32 @@
+# To be able to use this sample it is necessary to:
+# - Have a Pure backend with NVMe-RoCE support
+# - Have the Pure storage credentials in cinder-volume-pure-secrets.yaml
+# - Having nvme-fabrics kernel module loaded on the host (done by this sample)
```

Comment:
You should add into the docs about the nvme-rdma module that is needed along with nvme-fabrics? Does the `nvme-fabrics.yaml` you mention have nvme-fdma in as well? Given there are other drivers that need the `nvme-tcp` module as well, it might be adding all 3 kernel modules by default

Reply:
We have a `MachineConfig` that ensures that `nvme-fabrics` is loaded ([link](https://github.com/openstack-k8s-operators/cinder-operator/blob/main/config/samples/backends/bases/nvmeof/nvme-fabrics.yaml))
That file is added automatically because we have the line `  - ../../bases/nvmeof` in this Pure NVMe-RoCE backend's `kustomization.yaml` ([link](https://github.com/openstack-k8s-operators/cinder-operator/blob/8ccadb31ea95f030c1fae434db0145cd7c379dcd/config/samples/backends/pure/nvme-roce/kustomization.yaml#L2))
We don't need to manually load the kernel modules for the individual NVMe transport protocols, they are automatically loaded on usage by the `nvme-fabrics` one.  I tested this with the LVM nvme-tcp example ([link](https://github.com/openstack-k8s-operators/cinder-operator/tree/main/config/samples/backends/lvm/nvme-tcp)).
I could be wrong and it only works when using `nvmet` (like we do in LVM) but I don't think that's the case.

Reply:
Just to be in the safe side I have added a patch to the PR that loads the other nvme kernel modules supported by openstack: https://github.com/openstack-k8s-operators/cinder-operator/pull/259/commits/c076183ffae963f10bd1c2d722842a344a7c0406